PIETOOLS/PIETOOLS_demos/DEMO1_Simple_Stability_Simulation_and_Control.m

```diff
@@ -1,77 +1,102 @@
-z=state('out',2);
-%% Create the system 
-% Here we use a sys() class object to store the equations that describe
-% the PDE model. First, we initialize the object and then equations are
-% added to the object using addequation() function. Finally, setControl()
-% function is used designate u as a control input.
+% Declare independent variables
+pvar t s;   
+% Declare state, input, and output variables
+phi = state('pde',2);   x = state('ode');
+w = state('in');        u = state('in');
+z = state('out',2);
+% Declare the PDE
 odepde = sys();
 eq_dyn = [diff(x,t,1)==-x+u
-                diff(phi,t,1)==[0 1; c 0]*diff(phi,s,1)+[0;s]*w+[0 0;0 -b]*phi];
-eq_out= z ==[int([1 0]*phi,s,[0,1])
-                                 u];
+          diff(phi,t,1)==[0 1; c 0]*diff(phi,s,1)+[0;s]*w+[0 0;0 -b]*phi];
+eq_out= z ==[int([1 0]*phi,s,[0,1]); u];
 odepde = addequation(odepde,[eq_dyn;eq_out]);
-odepde= setControl(odepde,[u]); % set the control signal
-bc1 = [0 1]*subs(phi,s,0)==0; % add the boundary conditions
+odepde= setControl(odepde,u);   % set the control signal
+bc1 = [0 1]*subs(phi,s,0)==0;   % add the boundary conditions
 bc2 = [1 0]*subs(phi,s,1)==x;
 odepde = addequation(odepde,[bc1;bc2]);
-%% Simulating the Open Loop system
-% Here we simulate the PDE system without any control input to see the
-% open-loop behaviour of the system using PIESIM. The following options are
-% used to define simulation parameters, initial conditions and disturbance
-% inputs for the simulation. 
+
+% Convert to PIE
+pie = convert(odepde);
+
+
+% =============================================
+% === Declare the LPI
+
+% Run pre-defined stability executive
+[prog, P] = lpiscript(pie,'stability','light');
+
+% Run pre-defined L2-gain executive
+[prog, P, gam] = lpiscript(pie,'l2gain','light');
+
+% Run pre-defined controller synthesis executive
+[prog, Kval, gam_CL] = lpiscript(pie,'hinf-controller','light');
+% Build closed-loop PIE with optimal controller
+PIE_CL = closedLoopPIE(pie,Kval);
+PIE_CL = pie_struct(PIE_CL);
+PIE_CL = initialize(PIE_CL);
+
+
+% =============================================
+% === Simulate the system
+
+% Set PIESIM simulation options
 opts.plot = 'no';   % Do not plot the final solution
 opts.N = 8;         % Expand using 8 Chebyshev polynomials
-opts.tf = 10;        % Simulate up to t = 1;
-opts.dt = 1e-2;     % Use time step of 10^-3
-opts.intScheme=1;   % Time-step using Backward Differentiation Formula (BDF)
+opts.tf = 10;       % Simulate up to t = 10;
+opts.dt = 1e-2;     % Use time step of 10^-2
+opts.intScheme = 1; % Time-step using Backward Differentiation Formula (BDF)
 ndiff = [0,2,0];    % The PDE state involves 2 first order differentiable state variables   
 
-% To simulate the solution to the PDE system without controller we can use
-% the following code. 
+% Set initial conditions and disturbance for simulation
+syms st sx;
 uinput.ic.PDE = [0,0];  
 uinput.ic.ODE = 0;  
-uinput.u=0;
+uinput.u = 0;
```

Comment:
seems that PIESIM now requires 0*st and 0*sx for uinput to be symbolic. However this throws a warning...

Reply:
My issue with the the optimal controller synthesis does seem to be limited to this branch (it does not occur in master), so I'll need to fix that before merging.